src/extension/provider/launcherBeta.ts

```diff
@@ -0,0 +1,355 @@
+  }
+
+  get onDidChangeTreeData() {
+    return this._onDidChangeTreeData.event
+  }
+
+  async collapseAll() {
+    this.defaultItemState = TreeItemCollapsibleState.Collapsed
+    await commands.executeCommand('setContext', 'runme.launcher.isExpanded', false)
+    this._onDidChangeTreeData.fire(undefined)
+  }
+
+  async expandAll() {
+    this.defaultItemState = TreeItemCollapsibleState.Expanded
+    await commands.executeCommand('setContext', 'runme.launcher.isExpanded', true)
+    this._onDidChangeTreeData.fire(undefined)
+  }
+
+  getTreeItem(element: RunmeFile): TreeItem {
+    return element
+  }
+
+  async getChildren(element?: RunmeFile | undefined): Promise<RunmeFile[]> {
+    const allTasks = await this.tasks
+
+    if (!element) {
+      /**
+       * we need to tweak the folder name to force VS Code re-render the tree view
+       * see https://github.com/microsoft/vscode/issues/172479
+       */
+      ++sauceCount
+      return Promise.resolve(await this.getNotebooks(allTasks))
+    }
+
+    return Promise.resolve(await this.getCells(allTasks, element))
+  }
+
+  async getNotebooks(tasks: LoadEventFoundTask[]): Promise<RunmeFile[]> {
+    const foundTasks: RunmeFile[] = []
+    let prevDir: string | undefined
+
+    for (const task of tasks) {
+      const { documentPath } = task
+      const { outside, relativePath } = asWorkspaceRelativePath(documentPath)
+      if (outside) {
+        continue
+      }
+
+      if (prevDir === relativePath) {
+        continue
+      }
+
+      prevDir = relativePath
+      foundTasks.push(
+        new RunmeFile(`${relativePath}${sauceCount % 2 ? ' ' : ''}`, {
+          collapsibleState: this.defaultItemState,
+          lightIcon: 'tree-notebook.gif',
+          darkIcon: 'tree-notebook.gif',
+          contextValue: 'folder',
+        }),
+      )
+    }
+
+    return foundTasks
+  }
+
+  async getCells(tasks: LoadEventFoundTask[], element: RunmeFile): Promise<RunmeFile[]> {
+    const foundTasks: RunmeFile[] = []
+
+    let mdBuffer: Uint8Array
+    let prevFile: string | undefined
+    let notebook: Observable<NotebookData> | undefined
+
+    for (const task of tasks) {
+      const { documentPath, name, id, isNameGenerated } = task
+      const { outside, relativePath } = asWorkspaceRelativePath(documentPath)
+      if (outside || (!this.allowUnnamed && isNameGenerated)) {
+        continue
+      }
+
+      if (element.label.trimEnd() !== relativePath) {
+        continue
+      }
+
+      if (prevFile !== documentPath) {
+        prevFile = documentPath
+
+        try {
+          mdBuffer = await workspace.fs.readFile(Uri.parse(documentPath))
+        } catch (err: any) {
+          if (err.code !== 'FileNotFound') {
+            logger.error(`${err.message}`)
+          }
+          throw err
+        }
+
+        const token = new CancellationTokenSource().token
+        notebook = from(this.serializer.deserializeNotebook(mdBuffer, token))
```

Comment:
Again, why duplicate the work the task loader's already done? it's expensive to traverse plus deserialize files in a project/repo multiple times